### PR TITLE
Feat: Basic animation with cubes, core and lines

### PR DESCRIPTION
**Pull request by ZhichGaming**
The core is the 20-faced polyhedron frame.

This PR adds ThreeJS and Vite packages. It also removes React Strict Mode, which for some reason runs all constructors twice.

Removes all default padding and margin.

image

More details will be added to this animation in the future including particles, colored light, shaders and voice movement to replicate [this](https://www.youtube.com/watch?v=zZ9BppuxIiE) more closely.

### CODE REVIEW

**Comment by ZhichGaming** [comment]
image

File: src/Animation.tsx [hunk 1]
```diff
@@ -97,13 +108,17 @@ export default class Animation {
...
+            var cylinderGeometry = new THREE.CylinderGeometry(0.005, 0, 20, 32);
+            const material = new THREE.MeshPhongMaterial({ color: 0x0000ff, emissive: 0x0000ff, emissiveIntensity: 0.1 }); 
             const cylinder = new THREE.Mesh( cylinderGeometry, material );
 
             // Apply a random rotation to each line
             cylinder.applyMatrix4(this.getRandomRotationMatrix());
 
+            var angle = (i / this.numLines) * Math.PI * 2;
+            cylinder.position.x = Math.cos(angle) * 0.00001;
+            cylinder.position.y = Math.sin(angle) * 0.00001;
```
Comment (ZhichGaming) [inline src/Animation.tsx]: Why was this added? `cylinder.applyMatrix4(this.getRandomRotationMatrix());` already randomly generated an XYZ direction.

File: src/Animation.tsx [hunk 2]
```diff
@@ -82,6 +92,7 @@ export default class Animation {
     getRandomRotationMatrix() {
         var axis = new THREE.Vector3(Math.random(), Math.random(), Math.random()).normalize();
         var angle = Math.random() * Math.PI * 2;
+        console.log(axis, angle)
```
Comment (ZhichGaming) [inline src/Animation.tsx]: Please remove debug log statements.

File: src/Animation.tsx [hunk 3]
```diff
@@ -135,13 +150,14 @@ export default class Animation {
...
+        this.wireframe.rotation.x += 0.01
+        this.wireframe.rotation.y += 0.01
         this.core.rotation.x += 0.01
         this.core.rotation.y += 0.01
 
         this.lines.forEach(function (line) {
-            // line.rotation.x += 0.01;
-            // line.rotation.y += 0.01;
             line.rotation.z += 0.01;
+            console.log(line.rotation.z)
```
Comment (ZhichGaming) [inline src/Animation.tsx]: Debug statement.

File: src-tauri/tauri.conf.json [hunk 4]
```diff
@@ -6,7 +6,7 @@
     "distDir": "../dist"
   },
   "package": {
-    "productName": "Project-Raphael",
+    "productName": "Project Raphael",
```
Comment (ZhichGaming) [inline src-tauri/tauri.conf.json:9]: I might be wrong, but product names shouldn't have spaces in them because some programs don't accept spaces in a path name. We can keep "Project Raphael" for now and change it if we find an issue, this is only a comment.

**Review by ZhichGaming — COMMENTED** [review]
I like the the addition of green lights projecting on the lines although this was not in the [source](https://www.youtube.com/watch?v=zZ9BppuxIiE). We can make other changes in the future to improve this animation with [this task](https://github.com/users/ZhichGaming/projects/2/views/2?pane=issue&itemId=52841392). However, please provide a screenshot/video of the animation with the changes.

File: src/Animation.tsx (hunk 1 above)
Comment (ZhichGaming) [reply src/Animation.tsx]: I realized that this was actually setting its position. Why though?

Comment (Nvadidivan) [reply src/Animation.tsx]: Because if it's at 0, 0, 0. The light doesn't affect it

**Comment by Nvadidivan** [comment]
image

**Review by Nvadidivan — APPROVED** [review]
This is cool

**Review by DummyIsHere — APPROVED** [review]
All good
